### PR TITLE
Object results

### PR DESCRIPTION
**Pull request by mbenson**
This PR builds on https://github.com/RoadieHQ/roadie-backstage-plugins/pull/1053 which I hope will be accepted. I noticed that the result of the jsonata actions was documented as being of `object` type even though the Typescript seemed pretty adamant that the type was `string`. This PR adds an additional `as` input parameter which, when supplied the value `object` will set the action's result as the actual object returned from executing the JSONata `expression` against the specified `data` parameter, rather than encoding it back to the original format. Omitting `as` or specifying  `string` will trigger the original functionality.

#### :heavy_check_mark: Checklist

- [X] Added tests for new functionality and regression tests for bug fixes
- [X] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [X] Added or updated documentation (if applicable)

### CODE REVIEW

**Comment by mbenson** [comment]
Note that the effect of this PR could probably be achieved using multiple existing roadie utils actions (`fs:parse` + `jsonata`) in which case an alternative would be to correct the declared output type of the existing implementations to `string`.

File: plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/jsonata/json.ts [hunk 1]
```diff
@@ -44,6 +45,12 @@ export function createJsonJSONataTransformAction() {
             description: 'JSONata expression to perform on the input',
             type: 'string',
           },
+          as: {
+            title: 'Desired Result Type',
+            description:
+              'Permitted values are: "string" (default) and "object"',
+            type: 'string',
```
Comment (Xantier) [inline plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/jsonata/json.ts:52]: We could possibly add `enum: [the types]` here also?

File: plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/jsonata/json.ts [hunk 2]
```diff
@@ -64,26 +71,29 @@ export function createJsonJSONataTransformAction() {
...
           result: {
             title: 'Output result from JSONata',
-            type: 'object',
+            type: 'object | string',
           },
         },
       },
     },
     async handler(ctx) {
+      let post: (rz: any) => any;
```
Comment (Xantier) [inline plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/jsonata/json.ts:80]: maybe rename this to be a little bit more descriptive

Comment (mbenson) [reply plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/jsonata/json.ts:80]: Later I saw how the file formats on `fs:parse` are handled with a `Record` but I don't know that that would be particularly helpful in this scenario. I will expand on the variable name.

**Review by Xantier — APPROVED** [review]

**Pull request merged**